ui/StatusQ/src/StatusQ/Controls/StatusSwitch.qml

```diff
@@ -1,23 +1,34 @@
-import QtQuick 2.12
+import QtQuick 2.14
 import QtQuick.Controls 2.14
 import QtGraphicalEffects 1.14
 import StatusQ.Core 0.1
 import StatusQ.Core.Theme 0.1
 import StatusQ.Components 0.1
 
 Switch {
-    id: statusSwitch
+    id: root
 
-    indicator: Rectangle {
+    background: MouseArea {
+        cursorShape: enabled ? Qt.PointingHandCursor : Qt.ArrowCursor
+        acceptedButtons: Qt.NoButton
+    }
+
+    indicator: Item {
         id: oval
+
         implicitWidth: 52
         implicitHeight: 28
-        x: statusSwitch.leftPadding
+        x: root.leftPadding
         y: parent.height / 2 - height / 2
-        radius: 14
-        color: statusSwitch.checked ? Theme.palette.primaryColor1
-                               : Theme.palette.directColor8
 
+        Rectangle {
```

Comment:
can't this be set as the `contentItem` directly? or we need a wrapper `Item`?

Reply:
For now it's needed because according to the design `opacity` of the rectangle is semi-transparent when enabled. Having it as a top item, opacity set on it would affect child as well. From what I know the design may be adjusted in the future, then we can optimize it accordingly.